### PR TITLE
chore(storage-browser): grid updates

### PR DESCRIPTION
**Pull request by hbuchel**
#### Description of changes

- use auto rows so vertical gaps aren't present for empty rows
- use less columns

**Before**
Screenshot 2024-08-26 at 12 11 07 PM

**After**

Screenshot 2024-08-26 at 12 12 22 PM

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/styles/storage-browser.css [hunk 1]
```diff
@@ -3,29 +3,21 @@
   --storage-browser-gap: 0.6rem;
   display: grid;
   gap: var(--storage-browser-gap);
-  grid-template-columns: repeat(32, 1fr);
-  grid-template-rows: repeat(24, min-content);
-}
-
-.storage-browser__header {
```
Comment (hbuchel) [inline packages/react-storage/src/styles/storage-browser.css:10]: old css we're not using anymore

**Comment by calebpollman** [comment]
Will using removing the empty `rows` create a scenario where the space above the Table is inconsistent between Views?

**Comment by hbuchel** [comment]
> Will removing the empty `rows` create a scenario where the space above the Table is inconsistent between Views?

Sort of? In the current implementation, it does the same thing except the gaps from empty rows are still visible (so not the empty row itself, but the gap created by that row). This makes it so both the empty rows and the gaps from those rows are not visible which was creating weird inconsistent spacing.

**Review by timngyn — APPROVED** [review]
lgtm!

**Comment by hbuchel** [comment]
> > Will removing the empty `rows` create a scenario where the space above the Table is inconsistent between Views?
> 
> Sort of? In the current implementation, it does the same thing except the gaps from empty rows are still visible (so not the empty row itself, but the gap created by that row). This makes it so both the empty rows and the gaps from those rows are not visible which was creating weird inconsistent spacing.

Added a Before/After image to show the difference in gaps.

**Review by calebpollman — APPROVED** [review]

**Pull request merged**